trace/trace.go

```diff
@@ -76,6 +123,13 @@ func (t *Trace) Step(msg string, fields ...Field) {
 	t.steps = append(t.steps, traceStep{stepTime: time.Now(), msg: msg, fields: fields})
 }
 
+// Nest adds a new trace to the nestedTrace array
+func (t *Trace) Nest(msg string, fields ...Field) *Trace {
+	newTrace := New(msg, fields...)
+	t.nestedTrace = append(t.nestedTrace, newTrace)
+	return newTrace
+}
+
 // Log is used to dump all the steps in the Trace
```

Comment:
Expand this godoc a bit to clarify how nested traces are formatted in the output?